### PR TITLE
[MRG+1]: fix event sample indices read from eeglab files

### LINKED ISSUE #4944: eeglab latency to event without sfreq
when working on annotations to events I realize that the eeglab reader
convert `eeg.latency` to sample index with `int(latency)`

see: https://github.com/mne-tools/mne-python/blob/master/mne/io/eeglab/eeglab.py#L697

reading https://sccn.ucsd.edu/wiki/Chapter_03:_Event_Processing it seems that latency is in seconds to converting to events would require `sfreq`

@jona-sassenhagen @smgutstein  @mmagnuski thoughts? is this a bug?

### PR DESCRIPTION
**Pull request by mmagnuski**
fix event latency from matlab 1-based to python 0-based.
Fixes #4944.

### CODE REVIEW

File: mne/io/eeglab/tests/test_eeglab.py [hunk 1]
```diff
@@ -121,8 +121,11 @@ def test_io_set():
     shutil.copyfile(op.join(base_dir, 'test_raw.fdt'),
                     op.join(temp_dir, 'test_one_event.fdt'))
     event_id = {eeg.event[0].type: 1}
-    read_raw_eeglab(input_fname=one_event_fname, montage=montage,
-                    event_id=event_id, preload=True)
+    test_raw = read_raw_eeglab(input_fname=one_event_fname, montage=montage,
+                               event_id=event_id, preload=True)
+
+    # test that sample indices are read python-wise (zero-based)
+    assert mne.find_events(test_raw)[0, 0] == eeg.event[0].latency - 1
```
Comment (jasmainak) [inline mne/io/eeglab/tests/test_eeglab.py]: can you add a test to check they are all positive?

Comment (mmagnuski) [reply mne/io/eeglab/tests/test_eeglab.py]: sure, or do you prefer to throw an error **during reading** if some come out negative?

Comment (jasmainak) [reply mne/io/eeglab/tests/test_eeglab.py]: hmm ... I would think we have a bug if something comes out negative _during reading_

Comment (mmagnuski) [reply mne/io/eeglab/tests/test_eeglab.py]: Right, I'll add a ValueError to eeglab reader if latency comes out negative. I'm wondering if it is possible though due to how eeglab handles downsampling data (latency is then represented as float, so `int(0.7)` would be 0 and then -1 after index correction). But that is an issue for a next PR (for example rounding float latencies).

File: mne/io/eeglab/eeglab.py [hunk 2]
```diff
@@ -717,15 +717,15 @@ def _read_annotations_eeglab(eeg):
...
         if (len(onset) > 0) and hasattr(eeg.event[0], 'duration'):
             duration = [event.duration for event in eeg.event]
         else:
             duration = np.zeros(len(onset))
     else:
         # only one event - TypeError: 'mat_struct' object is not iterable
         description = [str(eeg.event.type)]
-        onset = [eeg.event.latency]
+        onset = [eeg.event.latency - 1]
         duration = getattr(eeg.event, 'duration', np.zeros(1))
```
Comment (agramfort) [inline mne/io/eeglab/eeglab.py]: as you can see there is something broken here as onset in Annotations should be in seconds but I will fix it in https://github.com/mne-tools/mne-python/pull/4936

**Comment by agramfort** [comment]
can you set the PR to MRG and update what's new? thanks

File: mne/io/eeglab/eeglab.py [hunk 3]
```diff
@@ -717,15 +717,15 @@ def _read_annotations_eeglab(eeg):
         description = []
     elif isinstance(eeg.event, np.ndarray):
         description = [str(event.type) for event in eeg.event]
-        onset = [event.latency for event in eeg.event]
+        onset = [event.latency - 1 for event in eeg.event]
```
Comment (jona-sassenhagen) [inline mne/io/eeglab/eeglab.py]: Maybe add a comment here, just to be sure?

**Comment by jona-sassenhagen** [comment]
Looks good to me.

File: mne/io/eeglab/eeglab.py (hunk 2 above)
Comment (mmagnuski) [reply mne/io/eeglab/eeglab.py]: Ah, right. Ok, so I'll leave it the way it is now here.

File: mne/io/eeglab/eeglab.py (hunk 3 above)
Comment (mmagnuski) [reply mne/io/eeglab/eeglab.py]: see a comment by @agramfort, few lines below, I'll leave it as it is for now.

**Comment by mmagnuski** [comment]
Last change was only whats_new so the CIs should come back green.

**Comment by mmagnuski** [comment]
@agramfort - one last thing, @jasmainak asked me to throw an error when any time sample becomes negative during reading - should I do it here or do you prefer to deal with it in your PR?

**Comment by mmagnuski** [comment]
Ok, I've added that ValueError - although I am wondering if it shouldn't be just a warning (for example we only give warnings about boundary events). Throwing error in the function that reads events would make it harder for someone to investigate what is wrong with the event latencies (but we can advise using sicpy.io.loadmat then) unless one has access to matlab on the same computer.
I've also added a test, but it will probably go back red as I have problems running tests locally :(

**Comment by jasmainak** [comment]
They can just use postmortem debugging to investigate. And if they give us a reproducible script, it's not too hard :)

**Comment by jasmainak** [comment]
@mmagnuski +1 for merge from me after circle is happy :)

**Comment by mmagnuski** [comment]
@jasmainak Ok, I finally corrected the test errors and one pep8 violation. My bold prediction is that CIs will be green finally. @agramfort @jona-sassenhagen I'm ready for last checks if you want to take another look.

**Comment by agramfort** [comment]
+1 for MRG

**Comment by mmagnuski** [comment]
🥗 📗 all green 💚 🍏

**Comment by larsoner** [comment]
Thanks @mmagnuski

**Comment by mmagnuski** [comment]
Thanks! 🚀

**Comment by jasmainak** [comment]
Thanks @mmagnuski :-)

**Pull request merged**